### PR TITLE
replace the last applied managed cluster api server url when it changes

### PR DESCRIPTION
**Pull request by zhujian7**
issue reference: https://issues.redhat.com/browse/ACM-4647

### CODE REVIEW

File: pkg/controllers/clusterca/clusterca_controller.go [hunk 1]
```diff
@@ -56,50 +56,82 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
...
-			return clusterConfigs, true
+
+		if reflect.DeepEqual(cluConfig.CABundle, clusterinfoConfig.CABundle) && cluConfig.URL == clusterinfoConfig.URL {
+			return clusterConfigs, false, false
 		}
-		return clusterConfigs, false
+
+		clusterConfigs[i].URL = clusterinfoConfig.URL
+		clusterConfigs[i].CABundle = clusterinfoConfig.CABundle
+		return clusterConfigs, true, cluConfig.URL != clusterinfoConfig.URL
```
Comment (qiujian16) [inline pkg/controllers/clusterca/clusterca_controller.go:134]: cluConfig.URL != clusterinfoConfig.URL this seems buggy, what happens if updateMC succeed and updateMCI fails? I think you should always check if lastApplied equals clusterinfoConfig after updateMC succeeds

Comment (zhujian7) [reply pkg/controllers/clusterca/clusterca_controller.go]: fixed, will always check if lastApplied equals clusterinfoConfig no matter  managedCluster needs to update or not.

**Pull request merged**